emcc.py

```diff
@@ -394,18 +394,35 @@ def find_output_arg(args):
   """
   outargs = []
   specified_target = None
-  use_next = False
-  for arg in args:
-    if use_next:
-      specified_target = arg
-      use_next = False
-      continue
+
+  arg_count = len(args)
+  i = 0
+
+  while i < arg_count:
+    arg = args[i]
+
     if arg == '-o':
-      use_next = True
-    elif arg.startswith('-o'):
+      if i != arg_count - 1:
+        specified_target = args[i + 1]
+      i += 2
+      continue
+
+    if arg.startswith('-o'):
       specified_target = arg[2:]
+      i += 1
+      continue
+
+    outargs.append(arg)
+
+    if arg == '-mllvm' and i != arg_count - 1:
+      # Explicitly skip over -mllvm arguments and their values because their
+      # values could potentially start with -o and be confused for output file
+      # specifiers.
+      outargs.append(arg[i + 1])
```

Comment:
Should this be "args[i + 1]"? Looks like you are accidentally indexing into 'arg' here!